### ISSUE TITLE
[BUG] issue serializing a sequence of nodes

### ISSUE THREAD

**Issue by peterstadler** [eXist-db/exist#4702]
**Describe the bug**

When trying to serialize a sequence of nodes, only the first gets serialized properly and the following nodes get atomized.
`(<a>foo</a>, <b>bar</b>) => serialize()` produces `"<a>foo</a> bar"`  on current eXist 6.1.0. (Tested at http://exist-db.org/exist/apps/eXide/index.html)

**Expected behavior**

The output should be `"<a>foo</a><b>bar</b>"` 

**To Reproduce**

```Xquery
xquery version "3.1";

module namespace mytest="http://mytest";

declare namespace test="http://exist-db.org/xquery/xqsuite";

declare %test:assertEquals("<a>foo</a><b>bar</b>") function mytest:a() {
    (<a>foo</a>, <b>bar</b>) => serialize()
};

declare %test:assertEquals("<a>foo</a><b>bar</b>") function mytest:b() {
    document { <a>foo</a>, <b>bar</b> } => serialize()
};
```

**Context (please always complete the following information):**
 - OS:  macOS 12.6.2
 - eXist-db version: 6.1.0
 - Java 1.8.0_342

**Comment by line-o** [comment]
I was able to reproduce the issue.

**Comment by line-o** [comment]
This is a regression from 6.0.1

**Comment by line-o** [comment]
`((<a>foo</a>, <b>bar</b>) ! serialize(.)) => string-join('')` is a possible workaround for some cases.

**Comment by joewiz** [comment]
On first glance the idea of serializing `document { <a>foo</a>, <b>bar</b> }` strikes me as odd given that the "document" here is not well-formed, but upon reviewing the serialization spec, an implementation should have no trouble doing so. I should add that BaseX and Saxon return the expected results.

**Comment by peterstadler** [comment]
> On first glance the idea of serializing `document { <a>foo</a>, <b>bar</b> }` strikes me as odd given that the "document" here would not yield a well-formed XML document, but it is perfectly legal [XDM](https://www.w3.org/TR/xpath-datamodel-31/#DocumentNode) ("a Document Node may be empty, it may have more than one Element Node as a child..."), and upon reviewing the serialization spec, an implementation should have no trouble doing so. I should add that BaseX and Saxon return the expected results. 

Yes, you're right and the second test case not ideal. It was constructed late at night ...
A better (real world example) might feature a processing-instruction and/or a comment before the root element. 

I only added this second test case to demonstrate that the serialize function works well for single nodes.

**Comment by line-o** [comment]
This issue was probably introduced with d267acab

**Comment by line-o** [comment]
It is also not possible to set the item-separator to an empty string.
This is crucial as it is the default separator for eXist <=6.0.1 !

```
declare
    %test:assertEquals("aaabbb")
function ser:serialize-atomic-empty-separator() {
    let $nodes := ("aaa", "bbb")
    return
        fn:serialize($nodes, map {"item-separator": ""})
};
```

**Comment by line-o** [comment]
@peterstadler please be aware that the default item-separator is now `" "` when it effectively was `""` before.

So, if it is important for your application to have the serialised nodes without the extra space in between you have to set
`map{'item-separator': ''}` in the options map when serialising.

**Comment by vrazanajao** [comment]
Hi all,
I was about to post my (eventually!) first message on eXistdb slack when I saw this issue and https://github.com/eXist-db/exist/pull/4703. I clearly encounter a similar problem with the following quite simple lines of code :

```
xquery version "3.1";

let $data := <data>
                <row n="1"><item>a</item><item>b</item></row>
                <row n="2"><item>y</item><item>z</item></row>
            </data>

return serialize($data/row[1]/node())
```

Output in eXistdb 6.1.0: "<item>a</item> Some text b"
Output in  eXistdb 6.0.1: "<item>a</item>Some text<item>b</item>"

I need the serialization as in version 6.0.1 in order to populate an ACE Editor intended to edit XML code on a web page. 
Thanks,
Vincent

**Comment by ahenket** [comment]
This is unfortunately also a showstopper for us in going to version 6.1.0.

**Comment by line-o** [comment]
Maybe it helps if users affected by this upvote the PR that is fixing this and other issues with `serialize` #4703 
If you are able to checkout the PR, build it and test if it works and post your results, this would help even more.
